src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -739,6 +741,76 @@ class StringGenerator : public Generator<std::string> {
     const size_t _alphabetLength;
 };
 
+/** `{^ChooseFromDataset:{...}` */
+class RandomStringFromDataset : public Generator<std::string> {
+
+public:
+    RandomStringFromDataset(const Node& node, GeneratorArgs generatorArgs)
+        : _rng{generatorArgs.rng},
+          _id{generatorArgs.actorId},
+          _path{node["path"].maybe<std::string>().value()},
+          _pathLength{_path.size()}{}
+
+    void loadDataset() {
+      {
+        if (_pathLength <= 0) {
+            BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax(
+            "ChooseFromDataset requieres non-empty path"));
+            }
+
+        {
+            if (_all_datasets.count(_path) == 0){
+                _ifs.open(_path, std::ifstream::in);
+                if (_ifs.is_open()){
+                    while(std::getline(_ifs, _current_line))
+                    {
```

Comment:
Please ensure you run your changed files through `clang-format`. Annoyingly this isn't part of the toolchain and we have no CI checking for it.

```sh
brew install clang-format
clang-format -i ./src/**/DocumentGenerator.*
```

Reply:
I get not output, so I guess that means it is ok?

```
dev/genny [PERF-2998●] » clang-format -i ./src/**/DocumentGenerator.* 
dev/genny [PERF-2998●] » echo $? 
0
```